Introduction of `simdjson-go` dependency breaks cross-compilation

## Expected Behavior
Minio can be compiled with GOARCH=arm64, GOARCH=arm, etc. 

## Current Behavior
Compilation fails:

```
λ make crosscompile
Testing builds for OS/Arch: linux/arm64
-->     linux/arm64:github.com/minio/minio
# github.com/minio/simdjson-go
../../../go/pkg/mod/github.com/minio/simdjson-go@v0.1.2/stage1_find_marks.go:75:16: undefined: find_structural_bits_in_slice
../../../go/pkg/mod/github.com/minio/simdjson-go@v0.1.2/stage1_find_marks.go:87:17: undefined: find_structural_bits_in_slice
../../../go/pkg/mod/github.com/minio/simdjson-go@v0.1.2/stage2_build_tape.go:86:6: undefined: parse_string_simd_validate_only
../../../go/pkg/mod/github.com/minio/simdjson-go@v0.1.2/stage2_build_tape.go:104:7: undefined: parse_string_simd
../../../go/pkg/mod/github.com/minio/simdjson-go@v0.1.2/stage2_build_tape.go:119:29: undefined: parse_number_simd
```


## Steps to Reproduce (for bugs)

```
git clone https://github.com/minio/minio
cd minio
make crosscompile
```


## Context
Cross-compilation was working up until a few days ago when the new `simdjson-go` lib was introduced. Without this ability, Minio cannot be compiled to run on any ARM device (e.g., Raspberry Pi's, and many other SBCs). I've maintained an unofficial Docker image w/ 700k+ pulls for ~2 years to provide a multi-architecture compatible build; this change, however, has caused all non-amd64 builds to [fail](https://app.circleci.com/jobs/github/jessestuart/minio-multiarch/3217) with the message quoted above.

Interestingly, it seems the same error occurs in [Minio's TravisCI](https://travis-ci.org/minio/minio/jobs/650302775)... but isn't flagged as a build failure:

![Screen Shot on 2020-02-15 at 20-11-35](https://user-images.githubusercontent.com/583147/74597567-80a1e500-502f-11ea-8a37-8c9ed616809c.png)

## Regression
Yes — https://github.com/minio/minio/pull/8401 appears to be the parent commit / PR.

--- a/buildscripts/cross-compile.sh
+++ b/buildscripts/cross-compile.sh
@@ -1,7 +1,8 @@
 #!/bin/bash
 
+set -e
 # Enable tracing if set.
-[ -n "$BASH_XTRACEFD" ] && set -ex
+[ -n "$BASH_XTRACEFD" ] && set -x
 
 function _init() {
     ## All binaries are static make sure to disable CGO.

--- a/go.mod
+++ b/go.mod
@@ -49,7 +49,7 @@ require (
 	github.com/hashicorp/vault/api v1.0.4
 	github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
 	github.com/jonboulle/clockwork v0.1.0 // indirect
-	github.com/json-iterator/go v1.1.7
+	github.com/json-iterator/go v1.1.9
 	github.com/klauspost/compress v1.9.8
 	github.com/klauspost/cpuid v1.2.2
 	github.com/klauspost/pgzip v1.2.1
@@ -70,7 +70,7 @@ require (
 	github.com/minio/minio-go/v6 v6.0.45
 	github.com/minio/parquet-go v0.0.0-20200125064549-a1e49702e174
 	github.com/minio/sha256-simd v0.1.1
-	github.com/minio/simdjson-go v0.1.2
+	github.com/minio/simdjson-go v0.1.3
 	github.com/minio/sio v0.2.0
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect

